src/cascadia/Remoting/Monarch.cpp

```diff
@@ -44,33 +46,48 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
     // - the ID assigned to the peasant.
     uint64_t Monarch::AddPeasant(Remoting::IPeasant peasant)
     {
-        // TODO:projects/5 This is terrible. There's gotta be a better way
-        // of finding the first opening in a non-consecutive map of int->object
-        const auto providedID = peasant.GetID();
-
-        if (providedID == 0)
-        {
-            // Peasant doesn't currently have an ID. Assign it a new one.
-            peasant.AssignID(_nextPeasantID++);
-        }
-        else
+        try
         {
-            // Peasant already had an ID (from an older monarch). Leave that one
-            // be. Make sure that the next peasant's ID is higher than it.
-            _nextPeasantID = providedID >= _nextPeasantID ? providedID + 1 : _nextPeasantID;
-        }
+            // TODO:projects/5 This is terrible. There's gotta be a better way
+            // of finding the first opening in a non-consecutive map of int->object
+            const auto providedID = peasant.GetID();
 
-        auto newPeasantsId = peasant.GetID();
-        _peasants[newPeasantsId] = peasant;
+            if (providedID == 0)
+            {
+                // Peasant doesn't currently have an ID. Assign it a new one.
+                peasant.AssignID(_nextPeasantID++);
+            }
+            else
+            {
+                // Peasant already had an ID (from an older monarch). Leave that one
+                // be. Make sure that the next peasant's ID is higher than it.
+                _nextPeasantID = providedID >= _nextPeasantID ? providedID + 1 : _nextPeasantID;
+            }
 
-        // Add an event listener to the peasant's WindowActivated event.
-        peasant.WindowActivated({ this, &Monarch::_peasantWindowActivated });
+            auto newPeasantsId = peasant.GetID();
+            // Add an event listener to the peasant's WindowActivated event.
+            peasant.WindowActivated({ this, &Monarch::_peasantWindowActivated });
 
-        // TODO:projects/5 Wait on the peasant's PID, and remove them from the
-        // map if they die. This won't work great in tests though, with fake
-        // PIDs.
+            _peasants[newPeasantsId] = peasant;
 
-        return newPeasantsId;
+            TraceLoggingWrite(g_hRemotingProvider,
+                              "Monarch_AddPeasant",
+                              TraceLoggingUInt64(providedID, "providedID", "the provided ID for the peasant"),
+                              TraceLoggingUInt64(newPeasantsId, "peasantID", "the ID of the new peasant"),
+                              TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE));
+            return newPeasantsId;
+        }
+        catch (...)
+        {
+            TraceLoggingWrite(g_hRemotingProvider,

```

Comment:
the fallback error handler should deal with this, but.. wecan do it ourselves too :)